packages/squiggle-lang/src/js/Distribution.ts

```diff
@@ -0,0 +1,105 @@
+import * as RSDistribution from "../rescript/ForTS/ForTS_Distribution/ForTS_Distribution.gen";
+import { distributionTag as Tag } from "../rescript/ForTS/ForTS_Distribution/ForTS_Distribution_tag";
+import { environment } from "../rescript/ForTS/ForTS__Types.gen";
+import { DistributionError } from "./DistributionError";
+import { wrapPointSetDist } from "./PointSetDist";
+import { resultMap2 } from "./types";
+
+type T = RSDistribution.distribution;
+export { Tag };
+
+export const wrapDistribution = (value: T): Distribution => {
+  const tag = RSDistribution.getTag(value);
+
+  return new tagToClass[tag](value);
+};
+
+abstract class AbstractDistribution {
+  abstract tag: Tag;
+  _value: T;
+
+  constructor(value: T) {
+    this._value = value;
+  }
+
+  pointSet(env: environment) {
+    const innerResult = RSDistribution.toPointSet(this._value, env);
+    return resultMap2(
+      innerResult,
+      wrapPointSetDist,
+      (v: RSDistribution.distributionError) => new DistributionError(v)
+    );
+  }
+
+  toString() {
+    RSDistribution.toString(this._value);
+  }
+
+  mean(env: environment) {
```

Comment:
Btw, we really don't need most of these methods in the 0.4.0-dev version, if it will be more work to finish them.

Reply:
`mean` is used in summary tables, and it wasn't that hard to implement (though it required some figuring out together with Umur).

I haven't converted anything that's not used in components.